packages/typespec-client-generator-core/src/interfaces.ts

```diff
@@ -114,6 +99,63 @@ export interface SdkOperationGroup {
   service: Namespace;
 }
 
+export type AccessFlags = "internal" | "public";
+
+/**
+ * This enum represents the different ways a model can be used in a method.
+ */
+export enum UsageFlags {
+  None = 0,
+  Input = 1 << 1,
+  Output = 1 << 2,
+  ApiVersionEnum = 1 << 3,
+  // Input and Json will also be set when JsonMergePatch is set.
+  JsonMergePatch = 1 << 4,
+  // Input will also be set when MultipartFormData is set.
+  MultipartFormData = 1 << 5,
+  // Used in spread.
+  Spread = 1 << 6,
+  /**
+   * @deprecated Use `Exception` instead.
+   */
+  // Output will also be set when Error is set.
+  Error = 1 << 7,
+  // Set when type is used in conjunction with an application/json content type.
+  Json = 1 << 8,
+  // Set when type is used in conjunction with an application/xml content type.
+  Xml = 1 << 9,
+  // Set when type is used for exception output.
+  Exception = 1 << 10,
+  // Set when type is used as LRO initial response.
+  LroInitial = 1 << 11,
+  // Set when type is used as LRO polling response.
+  LroPolling = 1 << 12,
+  // Set when type is used as LRO final envelop response.
+  LroFinalEnvelope = 1 << 13,
+}
+
+/**
+ * Flags used to indicate how a client is initialized.
+ * `Individually` means the client is initialized individually.
+ * `Parent` means the client is initialized by its parent.
+ */
+export enum InitializedByFlags {
+  Individually = 1 << 0,
+  Parent = 1 << 1,
+}
+
+/**
+ * Options used to indicate how to initialize a client.
+ * `parameters` is a model that used to .
+ * `initializedBy` is a flag that indicates how the client is initialized.
+ */
+export interface ClientInitializationOptions {
+  parameters?: Model;
+  initializedBy?: InitializedByFlags;
```

Comment:
Should we make `initializedBy` "required"?

Reply:
this is for decorator, not for type graph in tcgc.